File: mkindayzir/backend/app/schemas/work_item.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Any
from datetime import datetime


class WorkItemCreate(BaseModel):
    projectId: str
    type: str
    title: str
    description: Optional[str] = None
    priority: Optional[str] = None
    assigneeId: Optional[str] = None
    initiativeId: Optional[str] = None
    iterationId: Optional[str] = None
    parentId: Optional[str] = None
    storyPoints: Optional[int] = None
    dueDate: Optional[datetime] = None


class WorkItemUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    assigneeId: Optional[str] = None
    initiativeId: Optional[str] = None
    iterationId: Optional[str] = None
    parentId: Optional[str] = None
    storyPoints: Optional[int] = None
    dueDate: Optional[datetime] = None


class WorkItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    projectId: str
    number: int
    type: str
    title: str
    description: Optional[str] = None
    status: str
    priority: str
    assigneeId: Optional[str] = None
    reporterId: str
    initiativeId: Optional[str] = None
    iterationId: Optional[str] = None
    parentId: Optional[str] = None
    storyPoints: Optional[int] = None
    dueDate: Optional[str] = None
    resolvedAt: Optional[str] = None
    metadata: str
    position: int
    createdAt: str
    updatedAt: str
    deletedAt: Optional[str] = None

File: src/types/work-item.ts
import type { Project } from "./project";
import type { User } from "./user";

export interface WorkItem {
  id: string;
  projectId: string;
  number: number;
  type: string;
  title: string;
  description: string | null;
  status: string;
  priority: string;
  assigneeId: string | null;
  reporterId: string;
  initiativeId: string | null;
  iterationId: string | null;
  parentId: string | null;
  storyPoints: number | null;
  dueDate: string | null;
  resolvedAt: string | null;
  metadata: Record<string, unknown>;
  position: number;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
  project?: Project;
  assignee?: User;
  reporter?: User;
  initiative?: { id: string; name: string; projectId: string };
  iteration?: { id: string; name: string; projectId: string };
  labels?: Array<{ label: Label }>;
}

export interface Label {
  id: string;
  projectId: string;
  name: string;
  color: string;
  createdAt: string;
}

export interface WorkItemLink {
  id: string;
  sourceId: string;
  targetId: string;
  linkType: string;
  source?: WorkItem;
  target?: WorkItem;
  createdAt: string;
}

export interface WorkItemsResponse {
  workItems: WorkItem[];
  pagination: {
    page: number;
    limit: number;
    total: number;
    totalPages: number;
  };
}

export interface WorkItemComment {
  id: string;
  entityType: string;
  entityId: string;
  authorId: string;
  content: string;
  parentId: string | null;
  createdAt: string;
  updatedAt: string;
  author?: User;
}

export interface WorkItemActivity {
  id: string;
  entityType: string;
  entityId: string;
  userId: string;
  action: string;
  changes: Record<string, unknown> | null;
  createdAt: string;
  user?: User;
}

export interface WorkItemBulkUpdate {
  ids: string[];
  status?: string;
  priority?: string;
  assigneeId?: string | null;
  iterationId?: string | null;
  initiativeId?: string | null;
  labelIds?: string[];
}
